dace/transformation/subgraph/subgraph_fusion.py

```diff
@@ -527,6 +658,40 @@ def redirect(redirect_node, original_node):
                                         and sdfg.data(data).transient \
                                         and data not in do_not_override \
                                   for data in data_intermediate}
+        return subgraph_contains_data
+
+    def clone_intermediate_nodes(self, sdfg: dace.sdfg.SDFG,
+                                 graph: dace.sdfg.SDFGState,
+                                 intermediate_nodes: List[nodes.AccessNode],
+                                 out_nodes: List[nodes.AccessNode],
+                                 map_entries: List[nodes.MapEntry],
+                                 map_exits: List[nodes.MapExit]):
+        ''' Creates cloned access nodes and data arrays for nodes that are both in intermediate nodes 
+        and out nodes, redirecting output from the original node to the cloned node. Operates in-place.
+        :param sdfg: SDFG 
+        :param state: State of interest 
+        :param intermediate_nodes: List of intermediate nodes appearing in a fusible subgraph 
+        :param out_nodes: List of out nodes appearing in a fusible subgraph
+        :param map_entries: List of outermost scoped map entries in the subgraph 
+        :param map_exits: List of map exits corresponding to map_entries in order 
+        :return: A dict that maps each intermediate node that also functions as an out node 
+                       to the respective cloned transient node 
+        '''
+        def redirect(redirect_node, original_node):
```

Comment:
function in function is a bit weird, could this be moved out and prefixed with underscore?

Reply:
inlined